ci/utils.go

```diff
@@ -3,174 +3,255 @@ package ci
-	cmd.Flags().String("foundAmount", foundAmount, "Amount to fund wallet")
-	cmd.Flags().String("rpc", rpc, "RPC endpoint")
-	cmd.Flags().String("transferedAmount", transferedAmount, "Amount to be transfered")
-	cmd.Flags().String("toAddress", toAddress, "Transfer receiving address")
-
-	executeCommand(t, cmd, map[string]string{"chain": chain, "rpc": rpc})
-	return balanceStr
+func getTxTransfer(ctx context.Context) (string, error) {
+	xcFactory, ok := ctx.Value(xcFactoryKey).(*factory.Factory)
+	if !ok {
+		return "", fmt.Errorf("xcFactory not found in context")
+	}
+
+	chainConfig, ok := ctx.Value(chainConfigKey).(*xc.ChainConfig)
+	if !ok {
+		return "", fmt.Errorf("chainConfig not found in context")
+	}
+
+	toTransferAddress, err := getToTransferAddress(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	toTransferAmmount := "0.001"
+	timeout := time.Duration(60000000000)
+
+	fmt.Println(toTransferAddress + " " + toTransferAmmount)
+
+	args := []string{toTransferAddress, toTransferAmmount}
+	txTransfer, err := handler.TxTransferHandler(xcFactory, chainConfig, "", "", "", timeout, args)
+	if err != nil {
+		return "", err
+	}
+
+	return string(txTransfer), nil
 }
 
-func getTxTransfer(t *testing.T, chain, toAddress, transferedAmount string) string {
-	var txTransferStr string
-	cmd := &cobra.Command{
-		Use:   "transfer <to> <amount>",
-		Short: "Create and broadcast a new transaction transferring funds. The amount should be a decimal amount",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx := setupContext(t, chain, rpc)
-			cmd.SetContext(ctx)
-
-			args = append(args, toAddress, transferedAmount)
-			txTransfer, err := handler.TxTransferHandler(cmd, args)
-			require.NoError(t, err, "Failed to transfer tokens")
-			txTransferStr = string(txTransfer)
-			return nil
-		},
-	}
-
-	cmd.Flags().String("chain", chain, "Blockchain chain to use")
-	cmd.Flags().String("foundAmount", foundAmount, "Amount to fund wallet")
-	cmd.Flags().String("transferedAmount", transferedAmount, "Amount to be transfered")
-	cmd.Flags().String("rpc", rpc, "RPC endpoint")
-	cmd.Flags().String("toAddress", toAddress, "Transfer receiving address")
-	cmd.Flags().String("contract", contract, "Asset contract")
-	cmd.Flags().String("decimals", decimals, "Amount decimals")
-	cmd.Flags().String("memo", memo, "Memo")
-	cmd.Flags().Duration("timeout", timeout, "Timeout")
-
-	executeCommand(t, cmd, map[string]string{"toAddress": toAddress, "transferedAmount": transferedAmount, "chain": chain, "rpc": rpc})
-	return txTransferStr
+func getToTransferAddress(ctx context.Context) (string, error) {
+	randomBytes := make([]byte, 32)
+	_, err := rand.Read(randomBytes)
+	if err != nil {
+		return "", fmt.Errorf("Failed to generate random bytes: %v", err)
+	}
+
+	// Save wallet PRIVATE_KEY to reset it after genereting new wallet address.
+	walletPKey := os.Getenv("PRIVATE_KEY")
+
+	privateKey := hex.EncodeToString(randomBytes)
+	os.Setenv("PRIVATE_KEY", privateKey)
```

Comment:
I feel like we have three ways of passing arguments to functions:
* concrete type
* embedded in `Context`
* setting environment variables to be read out later

It would be best if this could be organized in a way where we only use concrete types.

We use the `Context` for the CLI plumbing - not applicable here.
We use environment variables to read inputs from user to a CLI program - not applicable here.

Unless there's a benefit to using `Context` or environment variables here that I'm missing, should stick to concrete types.

Reply:
Reworked it:
- only using concrete types/ normal arguments. no context embedded things or environment variables set up.
- removed all CLI plumbing